### PR TITLE
feat: Add ability to run on single notebook

### LINKED ISSUE #31: Support ability to run blackbook over a single file
Currently it seems that there is no way to run `blackbook` over a specific file path. Attempting to format a single notebook at path `path/to/notebooks/Example-Notebook.ipynb` fails to do anything

```
$ pip freeze | grep blackbook
blackbook==0.0.5
$ blackbook path/to/notebooks/Example-Notebook.ipynb 
2019-11-29 14:00:14.403 | INFO     | blackbook.__main__:main:25 - All done! 📖
2019-11-29 14:00:14.403 | INFO     | blackbook.__main__:main:27 - 0 notebooks reformatted. 0 left unchanged.
```

where running

```
$ blackbook path/to/notebooks/
2019-11-29 14:05:26.204 | INFO     | blackbook.__main__:main:25 - All done! 📖
2019-11-29 14:05:26.205 | INFO     | blackbook.__main__:main:27 - 1 notebooks reformatted. 0 left unchanged.
```

It would be nice to add support for targeting a single file, in the same way that Black does.

### PR DESCRIPTION
**Pull request by matthewfeickert**
Resolves #31 

Add ability to specify the path of a single Jupyter notebook and then format only that notebook. Additionally, check before doing anything that the path given actually is a valid path. Additionally, add tests.

```
* Add check that given path is valid
* Allow for path to a single Jupyter notebook to be used for formatting
* Add tests for invalid path and for single notebook formatting
* Add example to README
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@drvinceknight this is ready for review. Let me know if there are any questions or requested changes.

File: tests/test_format_notebook_content.py [hunk 1]
```diff
@@ -20,6 +21,43 @@ def test_format_notebook_content():
...
+        [
+            copy_cell["source"] == source_cell["source"]
+            for copy_cell, source_cell in zip(copy_json["cells"], source_json["cells"])
+        ]
+    )
+
+    # Have to run blackbook main to show spaces_copy.ipynb is unaffected
+    blackbook.__main__.main(source_notebook_path)
+    output_json = json.loads(source_notebook_path.read_text())
+    check_copy_json = json.loads(copy_notebook_path.read_text())
```
Comment (Nikoleta-v3) [inline tests/test_format_notebook_content.py:47]: do you need to create `check_copy_json`? Why not just used `copy_json` that you defined above?

File: tests/test_format_notebook_content.py [hunk 2]
```diff
@@ -20,6 +21,43 @@ def test_format_notebook_content():
...
+    shutil.copy(str(copy_notebook_path), str(source_notebook_path))
+    pathlib.Path.unlink(copy_notebook_path)
```
Comment (Nikoleta-v3) [inline tests/test_format_notebook_content.py:49]: I understand that these two lines undo the changes to the unformatted notebook and delete the copy but it took me a while to figure that out 😅 I am not a huge fan of inline comments but maybe one is needed here.

I also suggest moving these two lines after the assert. That way the changes will be undone if the test passes 👍

File: tests/test_format_notebook_content.py (hunk 1 above)
Comment (matthewfeickert) [reply tests/test_format_notebook_content.py:47]: If

```python
blackbook.__main__.main(source_notebook_path)
```

failed to act on only the source notebook it is being run on, then the default behavior of `blackbook` would make it format _all_ notebooks in the directory. This is proof that the test isn't using information _before_ the formatting took place.

File: tests/test_format_notebook_content.py (hunk 2 above)
Comment (matthewfeickert) [reply tests/test_format_notebook_content.py:49]: > I am not a huge fan of inline comments but maybe one is needed here.

Sure, can do.

> I also suggest moving these two lines after the assert. That way the changes will be undone if the test passes

I don't think you want that. If the test fails then the notebook is in an altered state and you'd have to reset the repo before you could run tests again. The tests should be as atomic as possible.

File: tests/test_format_notebook_content.py (hunk 1 above)
Comment (Nikoleta-v3) [reply tests/test_format_notebook_content.py:47]: That's fantastic. Thank you for clarifying 😄

Comment (matthewfeickert) [reply tests/test_format_notebook_content.py:47]: Sure! :) Happy to.

**Comment by matthewfeickert** [comment]
Thanks for the quick review of this PR, @Nikoleta-v3. Can you let me know if you and your group need anything else on this before you merge it in?

**Comment by Nikoleta-v3** [comment]
No worries @matthewfeickert. 

We usually aim to have two approvals before merging prs. So currently waiting for someone else from the team to review 👍

**Comment by matthewfeickert** [comment]
> We usually aim to have two approvals before merging prs

Sounds good. We do the same thing for `pyhf`. :)

**Comment by drvinceknight** [comment]
Thanks for the contribution @matthewfeickert :+1:
